sample/src/main/java/dev/arkbuilders/sample/storage/StorageDemoFragment.kt

```diff
@@ -0,0 +1,120 @@
+package dev.arkbuilders.sample.storage
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.EditorInfo
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.fragment.app.DialogFragment
+import dev.arkbuilders.sample.R
+import dev.arkbuilders.sample.databinding.FragmentStorageDemoBinding
+import dev.arkbuilders.sample.extension.getAbsolutePath
+import java.io.File
+
+class StorageDemoFragment: DialogFragment() {
+
+    private val TAG = StorageDemoFragment::class.java.name
+
+    private lateinit var mBinding: FragmentStorageDemoBinding
+    private val mMap by lazy { mutableMapOf<String, String>() }
+
+    private val mSelectDirRequest = registerForActivityResult(ActivityResultContracts.OpenDocumentTree()) { uri ->
+        uri?.let {
+            // call this to persist permission across device reboots
+            context?.contentResolver?.takePersistableUriPermission(uri, Intent.FLAG_GRANT_READ_URI_PERMISSION)
+            val workingDir = uri.getAbsolutePath()
+            mBinding.btnWorkingDir.text = workingDir
+            refreshFilesTree()
+        }
+    }
+
+    private fun getCurrentAbsolutePath(): String {
+        return mBinding.btnWorkingDir.text.toString() + "/" + ".ark/" + mBinding.edtStoragePath.text.toString()
```

Comment:
Could you please elaborate this method? Perhaps give some examples. Looks like we can get the current absolute path with support method from Android API

Reply:
@hieuwu This method returns the absolute path to the user selected folder based on the current working directory.
User's selected folder is manually typed in by user.